### PR TITLE
secureContext

### PR DESCRIPTION
**Pull request by shileifu**

### CODE REVIEW

File: pkg/apis/starrocks/v1alpha1/starrockscluster_types.go [hunk 1]
```diff
@@ -40,6 +43,12 @@ type StarRocksClusterSpec struct {
 	StarRocksCnSpec *StarRocksCnSpec `json:"starRocksCnSpec,omitempty"`
 }
 
+//The start starrocks user info.
+type User struct {
+	Name               string                    `json:"name,omitempty"`
```
Comment (kevincai) [inline pkg/apis/starrocks/v1alpha1/starrockscluster_types.go:48]: use `UID`, hide PodSecurityContext from end user, operator will assemble the correct SecurityContext for PODs.

File: pkg/be_controller/be_pod.go [hunk 2]
```diff
@@ -217,6 +217,15 @@ func (be *BeController) buildPodTemplate(src *v1alpha12.StarRocksCluster, beconf
 		NodeSelector:                  beSpec.NodeSelector,
 	}
 
+	if src.Spec.User == nil {
+		podSpec.SecurityContext = &corev1.PodSecurityContext{
+			RunAsUser:  rutils.GetInt64ptr(1000),
+			RunAsGroup: rutils.GetInt64ptr(1000),
+		}
+	} else if src.Spec.User.Name != "root" {
```
Comment (kevincai) [inline pkg/be_controller/be_pod.go:225]: uid == 0 means root

**Review by kevincai — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -12,9 +12,9 @@ endif
 
 PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 ifeq (,$(GOBIN))
-GOBIN=$PROJECT_DIR
+GOBIN=$PROJECT_DIR/bin
 else
-$(shell mkdir -p $(GOBIN)/bin)
+$(shell mkdir -p $(GOBIN))
```
[the remark this answers is not in the document]
Comment (shileifu) [reply Makefile:17]: yes，mkdir make sure the directory always exists.

**Review by kevincai — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

File: dockerfiles/Dockerfile_be_ubuntu [hunk 4]
```diff
@@ -1,9 +1,9 @@
 FROM ubuntu:22.04 as base_amd64
-RUN useradd --create-home --no-log-init --shell /bin/bash starrocks
+RUN useradd --create-home --no-log-init --shell /bin/bash -u 1000 starrocks
```
Comment (kevincai) [inline dockerfiles/Dockerfile_be_ubuntu:2]: explicitly set gid as well by `-g 1000`

File: pkg/common/constant.go [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package common
+
+const (
+	DefaultRunAsUserId = 1000
```
Comment (kevincai) [inline pkg/common/constant.go:4]: DefaultRunAsUserId -> DefaultFsGroupId

**Pull request merged**